classes/controller/AdminController.php

```diff
@@ -995,13 +995,13 @@ public function postProcess()
                 $action = Tools::getValue('action');
                 // no need to use displayConf() here
                 if (!empty($action) && method_exists($this, 'ajaxProcess' . Tools::toCamelCase($action))) {
-                    Hook::exec('actionAdmin' . ucfirst($this->action) . 'Before', ['controller' => $this]);
-                    Hook::exec('action' . get_class($this) . ucfirst($this->action) . 'Before', ['controller' => $this]);
+                    Hook::exec('actionAdmin' . ucfirst($action) . 'Before', ['controller' => $this]);
```

Comment:
For safety, I recommend using `ucfirst($this->action ?? '')` instead.

Reply:
@leemyongpakvn don't focus on the deprecation notice. It's the tip of the iceberg. The code uses $this->action by mistake. It was probably pasted from the code a few lines below. Please check the whole method for better context.